src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -17,14 +17,20 @@
 /**
  * InputBag is a container for user input values such as $_GET, $_POST, $_REQUEST, and $_COOKIE.
  *
+ * @template TInput of string|int|float|bool|null
+ *
  * @author Saif Eddin Gmati <azjezz@protonmail.com>
  */
 final class InputBag extends ParameterBag
 {
     /**
      * Returns a scalar input value by name.
      *
-     * @param string|int|float|bool|null $default The default value if the input key does not exist
+     * @template TDefault of TInput|null
```

Comment:
I think this should be solved using a conditional return type instead of a `TDefault` generic (which might not be inferred properly):

```
@param TInput|null $default

@psalm-return ($default is null ? TInput|null : TInput)
```

We want the static analyzer to properly infer the return type, both when using an explicit default and when omitting it (in both psalm and phpstan).

Note that if we go with @nicolas-grekas's proposal allowing more flexible usages, my proposal need to be adapted or dropped (but testing the effectiveness of the inference in phpstan and psalm should still be done IMO)